compiler/src/iree/compiler/Dialect/Stream/IR/StreamInterfaces.h

```diff
@@ -16,8 +16,13 @@
 
 namespace mlir::iree_compiler::IREE::Stream {
 
+using AffinityAndOpPair = std::pair<AffinityAttr, Operation *>;
+
+// The function could be slow, if any data flow analysis is involved. Thus, the
+// API provides the batch mode.
 using ResolveLayoutAttrFn = std::function<LogicalResult(
-    AffinityAttr, Operation *, SetVector<Attribute> &)>;
+    ArrayRef<AffinityAndOpPair>,
```

Comment:
style: you can add argument names here to help readability and intellisense (e.g. `std::function<void(int a, int b, int c)>`)